frontend/src/components/GameDetails/GameDetailsHeader.js

```diff
@@ -86,8 +88,93 @@ const GameDetailsHeader = (props) => {
         if (props.data.involved_companies) {
             setGameDeveloper(props.data.involved_companies.filter((e) => e.developer == true)[0].company.name);
         }
+        (async () => {
+            getFavorites();
+         })();
+         
+         
+
+       
     }, []);
 
+    useEffect(() =>{
+        if (curFavs.length > 0){
+            for (let i = 0; i < curFavs.length; i++){
+                if (curFavs[i].id == props.data.id){
+                    setFavText("Remove from Favorites");
+                    break;
+                }
+                setFavText("Add to Favorites");
+            }
+        }
+        
+
+    }, [curFavs]);
+
+    async function getFavorites(){
+        let idToken = await getUserIdToken();
+        const url = 'http://localhost:5000/users/profile/favorites/?idToken=' + idToken;
+        fetch(
+            url,
+            {
+                credentials: 'include',
+            }
+        ).then((res) => res.json())
+            .then(
+                (result) => {
+                    setCurFavs(result);
+
+                   
+                },
+                (error) => {
+                    console.log(error);
+                    setCurFavs(error);
+                }
```

Comment:
setting error object to the current favorites

Reply:
set to null instead